### PR TITLE
Add distributed search example

### PR DESCRIPTION
**Pull request by pan-x-c**
## Description

Add a distributed search example

## Checklist

Please check the following items before code is ready to be reviewed.

- [x]  Code has passed all tests
- [x]  Docstrings have been added/updated in Google Style
- [x]  Documentation has been updated
- [x]  Code is ready for review

### CODE REVIEW

File: examples/distributed_search/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Multi-Agent Copilot Search
+
+## Introduction
+
+This example application converts the user's questions into keywords to call the search engine, and then crawls a series of web pages to find answers. It involves three types of Agents, namely the UserAgent for the user, the SearcherAgent responsible for searching, and the AnswererAgent responsible for retrieve web pages.
+
+There are many web page links returned by the search engine. To improve performance, multiple instances of AnswererAgent need to run together. However, with the traditional single-process mode, even if there are multiple AnswererAgent instances, they can only obtain web page and answer questions one by one.
+
+With AgentScope's distributed mode, you can automatically make these AnswererAgent instances run at the same time to improve performance.
+
```
Comment (DavdGao) [inline examples/distributed_search/README.md:10]: 1. Plz add a "prerequisite" section here to remind users to install agentscope in distributed mode ( if needed, other packages).
2. Add a note section to tell which models are already tested for this example. For other models, balabala...

File: examples/distributed_search/README.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
+# Multi-Agent Copilot Search
+
+## Introduction
+
+This example application converts the user's questions into keywords to call the search engine, and then crawls a series of web pages to find answers. It involves three types of Agents, namely the UserAgent for the user, the SearcherAgent responsible for searching, and the AnswererAgent responsible for retrieve web pages.
```
Comment (DavdGao) [inline examples/distributed_search/README.md:5]: Make it clear that what users can learn from this example, like

```
From this example, your can learn
- How to xxx
- How to xxx
- How to xxx
```

File: examples/distributed_search/README.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+python main.py --num-workers 10 --search-engine bing --api-key xxxxx --use-dist
+```
+
+And if you want to run the above case in a traditional single-process mode, you can use the following command.
+
+```shell
+python main.py --num-workers 10 --search-engine bing --api-key xxxxx
+```
+
+You can ask the same question in both modes to compare the difference in runtime.
```
Comment (DavdGao) [inline examples/distributed_search/README.md:63]: Maybe some comparison results here?

**Review by DavdGao — COMMENTED** [review]
plz see inline comments

**Review by DavdGao — COMMENTED** [review]
plz update both Chinese and English readme.md

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -29,6 +29,7 @@ repos:
                 pb2\.py$
                 | grpc\.py$
                 | ^docs
+                | ^examples
```
[the remark this answers is not in the document]
Comment (pan-x-c) [reply .pre-commit-config.yaml:32]: There are multiple `main.py` in examples, causing mypy error.

File: examples/distributed_search/main.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+            logger.chat(result)
+        end_time = time.time()
+        logger.chat(
+            Msg(
+                name="system",
+                role="system",
+                content=f"Completed in [{end_time - start_time:.2f}]s",
+            ),
+        )
+        msg = user_agent()
```
[the remark this answers is not in the document]
Comment (pan-x-c) [reply examples/distributed_search/main.py:77]: This example does not require history information

**Review by DavdGao — APPROVED** [review]
LGTM

**Pull request merged**